### PR TITLE
Setup AppMap

### PR DESCRIPTION
**Pull request by petecheslock**

### CODE REVIEW

File: .github/workflows/appmap-analysis.yml [hunk 1]
```diff
@@ -13,7 +13,24 @@ jobs:
...
+      - name: Enable appmap gem update
+        run: bundle config unset deployment    
+      - name: Install AppMap tools
+        id: install-appmap
+        uses: getappmap/install-action@v1
+        with:
+          project-type: bundler
+      - name: Commit changes
+        uses: EndBug/add-and-commit@v9
+        with:
```
Comment (kgilpin) [inline .github/workflows/appmap-analysis.yml:25]: At this point in the build, all changes to the repo should be committed. That will ensure that the .gitignore, Gemfile, and Gemfile.lock are committed. So you should be able to remove the `with:` option.

**Review by kgilpin — CHANGES_REQUESTED** [review]

File: .github/workflows/appmap-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: appmap-analysis
+
+on:
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (kgilpin) [inline .github/workflows/appmap-analysis.yml:10]: Looks like you need to specify the `ref` for the commit to work - https://github.com/EndBug/add-and-commit#working-with-prs

File: .github/workflows/appmap-analysis.yml [hunk 3]
```diff
@@ -24,6 +24,7 @@ jobs:
         uses: EndBug/add-and-commit@v9
         with:
           add: appmap.yml
+          push: origin HEAD:${{ github.head_ref }}
```
Comment (kgilpin) [inline .github/workflows/appmap-analysis.yml:27]: OK, maybe this is better than modifying the checkout?

Comment (kgilpin) [reply .github/workflows/appmap-analysis.yml:27]: Never mind :-) It's not working https://github.com/land-of-apps/rails-7-app/actions/runs/6433013707/job/17469149885?pr=1#step:6:63

File: .github/workflows/appmap-analysis.yml (hunk 2 above)
Comment (petecheslock) [reply .github/workflows/appmap-analysis.yml:10]: yea - just added in the latest commit.

**Pull request closed without merging**